teps/0100-embedded-taskruns-and-runs-status-in-pipelineruns.md

```diff
@@ -221,13 +221,28 @@ identify related objects.
 
 #### 1. Add Minimal Embedded Status
 
-We will introduce a new struct to hold references to child `TaskRuns` and `Runs`:
+We will introduce a new struct to hold references to child `TaskRuns` and `Runs`, and
+their corresponding `WhenExpressions` and `ConditionChecks`:
```

Comment:
a few quick comments:
1. could you please make it clear that `whenexpressions` and `conditionchecks` are already present in `PipelineRunTaskRunStatus`?
2. Runs don't currently have `conditionChecks`; I'm guessing this is because `PipelineRunRunStatus` was created after conditions were deprecated. Because `conditions` are deprecated, we may be able to remove `conditionchecks` from `childReference` if we do it in the correct order. see https://github.com/tektoncd/pipeline/issues/3377.
3. with the inclusion of "whenexpressions" the childReference is back to being a "childStatus".

Reply:
1: Will do!

2: Yeah, that's what I assumed was the case. I guess we could say "if you're using `minimal` for `embedded-status`, you shouldn't be using `conditions`, since that's going away before `full` `embedded-status` goes away"?

3: Yeah, in my work-in-progress, I'm actually calling the struct `ChildStatusReference` - it's got _some_ status in it alongside the reference.

Reply:
sounds good! I think we might have to include `conditionChecks` in the minimal status for v1beta1 only, making a note that they will be removed when support for `Conditions` is removed. This can just be empty for runs.

Reply:
👍

Reply:
Oh, and in my work-in-progress branch (which I'll be opening as a WIP implementation PR to pipeline after this PR gets merged), it's already not doing anything with `Conditions` for `Run`s. I'll mark the field as deprecated as well.

Reply:
Aaaaand PR is updated!